File: tests/optionals/extract-double.js
export function description_of_double$option$(option) {
  if (option.length !== 0) {
    const unwrapped = option[0];

    if (unwrapped !== null) {
      const doubleUnwrapped = unwrapped;

      if (doubleUnwrapped) {
        return "True";
      }

      return "False";
    }

    return "Inner None";
  }

  return "Outer None";
}

File: tests/optionals/extract-double.swift
public func description_of_double(option: Bool??) -> String {
    if let unwrapped = option {
        if let doubleUnwrapped = unwrapped {
            if doubleUnwrapped {
                return "True"
            }
            return "False"
        }
        return "Inner None"
    }
    return "Outer None"
}
